internal/manifest/report_test.go

```diff
@@ -267,6 +281,154 @@ var reportTests = []struct {
+	add: []sliceAndEntry{
+		{entry: sampleFile, slice: oneSlice},
+		{entry: sampleHardLink, slice: oneSlice}},
+	expected: map[string]manifest.ReportEntry{
+		"/example-file": {
+			Path:       "/example-file",
+			Mode:       0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			HardLinkID: 1,
+		},
+		"/example-hard-link": {
+			Path:       "/example-hard-link",
+			Mode:       0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			HardLinkID: 1,
+		},
+	},
+}, {
+	summary: "Hard link to symlink",
+	add: []sliceAndEntry{
+		{entry: sampleSymLink, slice: oneSlice},
+		{entry: sampleHardLinkSym, slice: oneSlice}},
+	expected: map[string]manifest.ReportEntry{
+		"/example-link": {
+			Path:       "/example-link",
+			Mode:       fs.ModeSymlink | 0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			Link:       "/base/example-file",
+			HardLinkID: 1,
+		},
+		"/example-hard-link-sym": {
+			Path:       "/example-hard-link-sym",
+			Mode:       fs.ModeSymlink | 0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			Link:       "/base/example-file",
+			HardLinkID: 1,
+		},
+	},
+}, {
+	summary: "Multiple hard links groups",
+	add: []sliceAndEntry{{
+		entry: sampleFile,
+		slice: oneSlice,
+	}, {
+		entry: sampleHardLink,
+		slice: oneSlice,
+	}, {
+		entry: fsutil.Entry{
+			Path:   "/base/another-file",
+			Mode:   0777,
+			SHA256: "another-file_hash",
+			Size:   5678,
+		},
+		slice: otherSlice,
+	}, {
+		entry: fsutil.Entry{
+			Path:     "/base/another-hard-link",
+			Mode:     0777,
+			Link:     "/base/another-file",
+			HardLink: true,
+		},
+		slice: otherSlice,
+	}},
+	expected: map[string]manifest.ReportEntry{
+		"/example-file": {
+			Path:       "/example-file",
+			Mode:       0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			HardLinkID: 1,
+		},
+		"/example-hard-link": {
+			Path:       "/example-hard-link",
+			Mode:       0777,
+			SHA256:     "example-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{oneSlice: true},
+			HardLinkID: 1,
+		},
+		"/another-file": {
+			Path:       "/another-file",
+			Mode:       0777,
+			SHA256:     "another-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{otherSlice: true},
+			HardLinkID: 2,
+		},
+		"/another-hard-link": {
+			Path:       "/another-hard-link",
+			Mode:       0777,
+			SHA256:     "another-file_hash",
+			Size:       5678,
+			Slices:     map[*setup.Slice]bool{otherSlice: true},
+			HardLinkID: 2,
+		},
+	},
+}, {
+	summary: "Hard links to same file in different slices",
+	add: []sliceAndEntry{{
+		entry: sampleFile, slice: oneSlice,
+	}, {
+		entry: sampleHardLink, slice: oneSlice,
```

Comment:
We don't need this one, right? We are testing that the slices do not have to match.

Reply:
Removed.